src/test/java/org/apache/sling/resourceresolver/impl/mapping/PagedQueryIteratorTest.java

```diff
@@ -106,23 +105,39 @@ public void testSimpleWrongOrder() {
         }
     }
 
-    @Ignore("SLING-12384: resources with empty keys lost")
     @Test
     public void testPagedWithEmpty() {
         String[] expected = new String[] { "", "a", "b", "c", "d" };
         Collection<Resource> expectedResources = toResourceList(expected);
         Collection<Resource> expectedFilteredResources = filter("", expectedResources);
         when(resourceResolver.findResources(eq("testPagedWithEmpty ''"), eq("JCR-SQL2")))
                 .thenReturn(expectedFilteredResources.iterator());
-        Iterator<Resource> it = mapEntries.new PagedQueryIterator("alias", PROPNAME, resourceResolver, "testPagedWithEmpty '%s'",
+        PagedQueryIterator it = mapEntries.new PagedQueryIterator("alias", PROPNAME, resourceResolver, "testPagedWithEmpty '%s'",
                 2000);
-        for (String key : expected) {
-            assertEquals(key, getFirstValueOf(it.next(), PROPNAME));
-        }
-        assertFalse(it.hasNext());
+        checkResult(it, expected);
+        assertEquals("", it.getWarning());
+    }
+
+    @Test
+    public void testPagedLargePage() {
+        String[] expected = new String[] { "a", "a", "a", "a", "a", "a", "a", "a", "a", "a", "a", "a", "a", "a", "a", "a", "a", "a",
```

Comment:
```suggestion
        String[] expected = new String[140];
        Arrays.fill(expected,"a");
```
(I haven't counted the instances of 'a', but I assume it's 140 because that number is mentioned below.)